### PR TITLE
Dependency download performance optimization

### PR DESCRIPTION
**Pull request by SergeShkurko**
Rewritten pull request (#27) with fix (#26)

### CODE REVIEW

**Comment by SergeShkurko** [comment]
@jonasfj, can you review pull request?

**Comment by jonasfj** [comment]
Aren't you missing the changes to `example/src/examples/cow_repository.dart`? Something that handles the `null` values?

**Comment by mit-mit** [comment]
@SergeShkurko did you get a chance to consider the question from jonas?

**Comment by jonasfj** [comment]
Fix the conflicts, then we can land this as it only touches the example code.

**Comment by SergeShkurko** [comment]
@jonasfj hello! I'm rename merge request and make optimizations for more faster dependency downloading. Can you review?

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,27 @@
-# Don’t commit the following directories created by pub.
-.dart_tool
-.pub/
+# Miscellaneous
```
Comment (kevmoo) [inline .gitignore]: Please revert the change to this file. Ignore your IDE files in `~/.gitignore_global`

Comment (SergeShkurko) [reply .gitignore]: corrected

**Comment by jonasfj** [comment]
Future reference, it's always better to file a new PR, if you're changing the nature of what the pull-request does...

I'm not sure I'm happy tweaking `.gitignore` or `pubspec.yaml`, why we need to touch these?

File: .gitignore [hunk 2]
```diff
@@ -3,4 +3,4 @@
 .pub/
 .packages
 packages
-pubspec.lock
+pubspec.lock
```
Comment (kevmoo) [inline .gitignore]: add back trailing newline...

Comment (SergeShkurko) [reply .gitignore]: corrected

**Comment by jonasfj** [comment]
@SergeShkurko, I'm curious what are you using this for?
If you're using this to run a proxy and copy-on-write pub server, I'm starting to think that it would be better to fork.

If you were to fork, I think we'd be happy to feature a link to your project. There is a few people interested in this sort of thing. And it seems unwise to me that we are gatekeepers for people developing and improving a copy-on-write proxy server for pub.

How do you feel about forking the entire repository and giving a new name, neat README, and dropping me a link to put into our README? :)
The README on this project says it's alpha and not intended for production use.

**Comment by mit-mit** [comment]
Closing this PR: This repo is not actively maintained
